com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -93,33 +93,24 @@ static CloudControllerClient getControllerClient(DelegateExecution context, Clou
     }
 
     public static String determineCurrentUser(VariableScope scope, StepLogger stepLogger) {
-        String userId = Authentication.getAuthenticatedUserId();
-        String previousUser = (String) scope.getVariable(Constants.VAR_USER);
-        // Determine the current user
-        if (userId != null && !userId.equals(previousUser)) {
-            stepLogger.debug(Messages.AUTHENTICATED_USER_ID, userId);
-            stepLogger.debug(Messages.PREVIOUS_USER, previousUser);
-        }
+        String userId = (String) scope.getVariable(Constants.VAR_USER);
         if (userId == null) {
-            // If the authenticated user cannot be determined,
-            // use the user saved by the previous service task
-            userId = previousUser;
-            if (userId == null) {
-                // If there is no previous user, this must be the first service task
-                // Use the process initiator in this case
-                userId = (String) scope.getVariable(Constants.PARAM_INITIATOR);
-                stepLogger.debug(Messages.PROCESS_INITIATOR, userId);
-                if (userId == null) {
-                    throw new SLException(Messages.CANT_DETERMINE_CURRENT_USER);
-                }
-            }
+            userId = getProcessInitiator(scope, stepLogger);

```

Comment:
This 'if' branching... will it ever be considered?
In what cases will the userId be null?
Previously that happened when there is no previous user and we are at the first service task.
Is this still possible?
What I am trying to ask is if this branching is necessary, in what scenario during the process will 'getVariable' for VAR_USER not be sufficient enough and need to use the INITIATOR?